src/nni_manager/training_service/reusable/environments/amlEnvironmentService.ts

```diff
@@ -0,0 +1,148 @@
+    }
+
+    public createEnviornmentInfomation(envId: string, envName: string): EnvironmentInformation {
+        return new AMLEnvironmentInformation(envId, envName);
+    }
+
+    public async config(key: string, value: string): Promise<void> {
+        switch (key) {
+            case TrialConfigMetadataKey.AML_CLUSTER_CONFIG:
+                this.amlClusterConfig = <AMLClusterConfig>JSON.parse(value);
+                break;
+
+            case TrialConfigMetadataKey.TRIAL_CONFIG: {
+                if (this.amlClusterConfig === undefined) {
+                    this.log.error('aml cluster config is not initialized');
+                    break;
+                }
+                this.amlTrialConfig = <AMLTrialConfig>JSON.parse(value);
+                // Validate to make sure codeDir doesn't have too many files
+                await validateCodeDir(this.amlTrialConfig.codeDir);
+                break;
+            }
+            default:
+                this.log.debug(`AML not proccessed metadata key: '${key}', value: '${value}'`);
+        }
+    }
+
+    public async refreshEnvironmentsStatus(environments: EnvironmentInformation[]): Promise<void> {
+        environments.forEach(async (environment) => {
+            const amlClient = (environment as AMLEnvironmentInformation).amlClient;
+                    if (!amlClient) {
+            throw new Error('AML client not initialized!');
+            }
+            const status = await amlClient.updateStatus(environment.status);
+            switch (status.toUpperCase()) {
+                case 'WAITING':
+                case 'RUNNING':
+                case 'QUEUED':
+                    // RUNNING status is set by runner, and ignore waiting status
+                    break;
+                case 'COMPLETED':
+                case 'SUCCEEDED':
+                    environment.setFinalStatus('SUCCEEDED');
+                    break;
+                case 'FAILED':
+                    environment.setFinalStatus('FAILED');
+                    break;
+                case 'STOPPED':
+                case 'STOPPING':
+                    environment.setFinalStatus('USER_CANCELED');
+                    break;
+                default:
+                    environment.setFinalStatus('UNKNOWN');
+            }
+        });
+    }
+
+    public async startEnvironment(environment: EnvironmentInformation): Promise<void> {
+        if (this.amlClusterConfig === undefined) {
+            throw new Error('AML Cluster config is not initialized');
+        }
+        if (this.amlTrialConfig === undefined) {
+            throw new Error('AML trial config is not initialized');
+        }
+        const amlEnvironment: AMLEnvironmentInformation = environment as AMLEnvironmentInformation;
+        const environmentLocalTempFolder = path.join(this.experimentRootDir, this.experimentId, "environment-temp");
+        environment.command = `import os\nos.system('${amlEnvironment.command}')`;
```

Comment:
Needs to escape special characters like `'` here.

Reply:
don't need to process this command, this is environment command, not trial's command here.